### ISSUE TITLE
Overloading and High memory consumption while downloading data 

### ISSUE THREAD

**Issue by sivico26** [sagnikbanerjee15/Finder#20]
Hi there, thanks for developing  `finder`

I am currently running `finder` on a fungus genome. I put several dozens of SRA libraries in the metadata file. The process is currently at the first step, downloading from SRA and aligning the data using STAR.

Currently, this is the landscape on the node I am running `finder`:

![image](https://user-images.githubusercontent.com/37163025/125994913-c4f43377-3d3a-470f-93b6-41823490b613.png)

The load average is very over the roof. Do you know why is this happening? Is this normal? Can you further explain to me what is `downloadAndDumpFastqFromSRA.py` doing besides prefetching and using `fasterq-dump`? It seems to me it is creating too many processes.

On the other hand, memory is another issue that I worry about. This should be a fairly small genome (~43 Mb), but it is using 380+ Gb of memory! it has even swapped already (which slows tremendously the process). I am not sure which processes are taking that much space. It is not clear to me that it is STAR, because if it were, memory should have been released after the alignment process. So I wonder why Memory is almost full _while_ downloading data. 

For reference, this is the command line I used:
```
finder -mf $folder/new_metadata.csv -out_dir $folder/output -g $folder/input/strain_masked.fasta -gdir_star $star_folder -gdir_olego $olego_folder --cpu $threads --addUTR
```

I would appreciate any thoughts on the matter. My TI admins are asking and I would like to be able to answer them.
Regards

**Comment by sagnikbanerjee15** [comment]
Hi @sivico26,

Thank you for your interest in `finder`. The `downloadAndDumpFastqFromSRA.py` script will spawn multiple processes for downloading the raw data. Even then I am not convinced that it should take such a large amount of memory since `prefetch` writes to the disk. You can rerun `finder` with `/usr/bin/time --verbose` option. It can tell you whether it is this process that uses such a large amount of RAM.

Thank you.

**Comment by sivico26** [comment]
Hi, thanks for the suggestion.

If I stop `finder` and launch it again with the files exactly as they were at that point, Would `finder` be smart enough to retake the process where he left it (let's say he already downloaded and aligned half of the data), or would it start from the beginning? I ask because it has not reached any checkpoint yet.

Regards

**Comment by sagnikbanerjee15** [comment]
Yes, `finder` will skip the steps that have already been computed.

Thanks.

**Comment by sivico26** [comment]
Just wanted to let you know I did not have the chance to apply your suggestion because the node failed during `finder` execution (It is currently down, so nothing I can do at the moment).
